src/blockrelay/graphene.cpp

```diff
@@ -477,26 +488,33 @@ bool CGrapheneBlock::HandleMessage(CDataStream &vRecv, CNode *pfrom, std::string
     return result;
 }
 
-void CGrapheneBlock::FillTxMapFromPools(std::map<uint64_t, uint256> &mapTxFromPools)
+void CGrapheneBlock::FillTxMapFromPools(std::map<uint64_t, CTransactionRef> &mapTxFromPools)
 {
+    {
+        boost::unique_lock<boost::mutex> lock(csCommitQ);
```

Comment:
just a (probably dumb) question, which is probably not related with this PR. I noticed that` csCommitQ` is always used via boost `unique_lock` rather than one of the macros  (eg `LOCK`) we define in `sync.h` ?

Reply:
I copied the lock code from mempool_sync.cpp, which itself copies from txadmission.cpp (also found in validation.cpp). As for the underlying reason, I am not exactly sure. I just tried using READLOCK instead, but received a compile error because apparently csCommitQ is of type `AnnotatedMixin<boost::mutex>` as opposed to `AnnotatedMixin<boost::shared_mutex>*`.  I guess I could create a shared pointer from csCommitQ and it should work? Does `boost::unique_lock<boost::mutex>` give you concern?

Reply:
Nope, I'm just trying to understand the locking code. Not saying that this is wrong on the contrary. Thanks for the chiming in.

Reply:
I needed Wait functionality -- its a CWaitableCriticalSection, not a CriticalSection.  We could create the same LOCK and deadlock detection infrastructure for this type as well, and we SHOULD if its ever used for more than this single object (whose locking is very low level, in the sense that it locks one and only one data structure).